CoreWebServer/src/au/gov/asd/tac/constellation/webserver/services/NewGraph.java

```diff
@@ -113,13 +116,23 @@ public void callService(final PluginParameters parameters, final InputStream in,
         final String graphName = SchemaFactoryUtilities.getSchemaFactory(schemaName).getLabel().trim().toLowerCase();
         GraphOpener.getDefault().openGraph(dualGraph, graphName);
 
-        final String newId = RestServiceUtilities.waitForGraphChange(existingId);
+        String newId = "";
 
-        final ObjectMapper mapper = new ObjectMapper();
-        final ObjectNode root = mapper.createObjectNode();
-        root.put("id", newId);
-        root.put("name", GraphNode.getGraphNode(newId).getDisplayName());
-        root.put("schema", schemaName);
-        mapper.writeValue(out, root);
+        try {
+            newId = RestServiceUtilities.waitForGraphChange(existingId).get(10, TimeUnit.SECONDS);
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        } catch (final ExecutionException | TimeoutException ex) {
+            throw new RestServiceException(ex);
+        }
+
+        if (!newId.isBlank() && !newId.isEmpty()) {
```

Comment:
`StringUtils.isNotBlank(newId)`

Side note: empty implies blank (i.e. the empty string is a subset of the blank strings which contain at most only whitespace characters) so you don't need to specify both